### PR TITLE
DLS-3548: Removed compilation warnings

### PR DESCRIPTION
**Pull request by rambabu-posa**

### CODE REVIEW

File: app/utils/CheckYourAnswersHelper.scala [hunk 1]
```diff
@@ -107,7 +104,4 @@ class CheckYourAnswersHelper(userAnswers: UserAnswers)(implicit messages: Messag
     }
 }
 
-object CheckYourAnswersHelper {
-
-  private val dateFormatter = DateTimeFormatter.ofPattern("d MMMM yyyy")
-}
+object CheckYourAnswersHelper
```
Comment (calinursaciuc) [inline app/utils/CheckYourAnswersHelper.scala:107]: I suppose that formatter is not used if deleted?

File: build.sbt [hunk 2]
```diff
@@ -35,10 +38,12 @@ lazy val root = (project in file("."))
     ScoverageKeys.coverageHighlighting := true,
     scalacOptions ++= Seq("-feature"),
     libraryDependencies ++= AppDependencies(),
+    libraryDependencies ++= Seq(
+      compilerPlugin("com.github.ghik" % "silencer-plugin" % "1.6.0" cross CrossVersion.full),
```
Comment (calinursaciuc) [inline build.sbt:42]: please don't have duplicate keys here and add all the dependencies inside AppDependencies. Also for what exactly is used this silencer?

**Review by calinursaciuc — CHANGES_REQUESTED** [review]
All dependencies to be in one place. Not sure what silencer does and it just falsely suppress warnings instead of fixing them.

File: app/utils/CheckYourAnswersHelper.scala (hunk 1 above)
Comment (rambabu-posa) [reply app/utils/CheckYourAnswersHelper.scala:107]: **We are using this `dateFormatter` in two places, but not sure whether we use them or not:**
```
https://github.com/hmrc/amls-estate-agency-business-frontend/blob/master/.g8/datePage/migrations/%24className__snake%24.sh#L60
https://github.com/hmrc/amls-estate-agency-business-frontend/blob/master/migrations/applied_migrations/DateOfChange.sh#L60
```

Comment (calinursaciuc) [reply app/utils/CheckYourAnswersHelper.scala:107]: Maybe is good to keep it for safety?

File: build.sbt (hunk 2 above)
Comment (calinursaciuc) [reply build.sbt:42]: These to be moved in AppDependencies

**Review by lutnos — APPROVED** [review]

**Pull request merged**